tests/Query/ConfigLoader.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace DoctrineExtensions\Tests\Query;
+
+use Doctrine\ORM\Configuration;
+use Symfony\Component\Yaml\Parser;
+
+class ConfigLoader
+{
+    const MYSQL    = 'mysql';
+    const ORACLE   = 'oracle';
+    const POSTGRES = 'postgres';
+    const SQLITE   = 'sqlite';
+
+    /**
+     * @param Configuration $configuration
+     * @param string        $database
+     */
+    public static function load(Configuration $configuration, $database)
+    {
+        $parser = new Parser();
+        // Load the corresponding config file.
+        $parsed = $parser->parse( file_get_contents( realpath( __DIR__ . '/../../config/' . $database . '.yml' ) ) );
+
+        // Make sure it has the correct structure.
+        if (array_key_exists( 'doctrine', $parsed)) {
+            if (array_key_exists('orm', $parsed['doctrine'])) {
+                if (array_key_exists('dql', $parsed['doctrine']['orm'])) {
+                    $parsed = $parsed['doctrine']['orm']['dql'];
+                } else {
+                    return;
+                }
+            } else {
+                return;
+            }
+        } else {
+            return;
+        }
```

Comment:
I think you may want to go for simplicity, a key error is what I would expect in this scenario, I don't think you need to do anything, just assume the keys are there and let it error out if they're not.